python/langsmith/_internal/_background_thread.py

```diff
@@ -102,41 +102,47 @@ def _tracing_thread_drain_queue(
-        return None, None
-    with client.compressed_traces.lock:
-        client.compressed_traces.compressor_writer.flush()
-        current_size = client.compressed_traces.buffer.tell()
+    try:
+        if client.compressed_traces is None:
+            return None, None
+        with client.compressed_traces.lock:
+            client.compressed_traces.compressor_writer.flush()
+            current_size = client.compressed_traces.buffer.tell()
 
-        pre_compressed_size = client.compressed_traces.uncompressed_size
+            pre_compressed_size = client.compressed_traces.uncompressed_size
 
-        if size_limit is not None and size_limit <= 0:
-            raise ValueError(f"size_limit must be positive; got {size_limit}")
-        if size_limit_bytes is not None and size_limit_bytes < 0:
-            raise ValueError(
-                f"size_limit_bytes must be nonnegative; got {size_limit_bytes}"
-            )
+            if size_limit is not None and size_limit <= 0:
+                raise ValueError(f"size_limit must be positive; got {size_limit}")
+            if size_limit_bytes is not None and size_limit_bytes < 0:
+                raise ValueError(
+                    f"size_limit_bytes must be nonnegative; got {size_limit_bytes}"
+                )
 
-        if (size_limit_bytes is None or current_size < size_limit_bytes) and (
-            size_limit is None or client.compressed_traces.trace_count < size_limit
-        ):
-            return None, None
+            if (size_limit_bytes is None or current_size < size_limit_bytes) and (
+                size_limit is None or client.compressed_traces.trace_count < size_limit
+            ):
+                return None, None
 
-        # Write final boundary and close compression stream
-        client.compressed_traces.compressor_writer.write(
-            f"--{_BOUNDARY}--\r\n".encode()
-        )
-        client.compressed_traces.compressor_writer.close()
+            # Write final boundary and close compression stream
+            client.compressed_traces.compressor_writer.write(
+                f"--{_BOUNDARY}--\r\n".encode()
+            )
+            client.compressed_traces.compressor_writer.close()
 
-        filled_buffer = client.compressed_traces.buffer
-        filled_buffer.context = client.compressed_traces._context
+            filled_buffer = client.compressed_traces.buffer
+            filled_buffer.context = client.compressed_traces._context
 
-        compressed_traces_info = (pre_compressed_size, current_size)
+            compressed_traces_info = (pre_compressed_size, current_size)
 
-        client.compressed_traces.reset()
+            client.compressed_traces.reset()
 
-    filled_buffer.seek(0)
-    return (filled_buffer, compressed_traces_info)
+        filled_buffer.seek(0)
+        return (filled_buffer, compressed_traces_info)
+    except Exception:
+        logger.error("Error in tracing queue", exc_info=True)
```

Comment:
Could we make  this more explicit so the user knows it isn't impacting their runtime, but rather indicates a langsmith trace may have failed to submit?

Want to avoid having people remove langsmith because they think it's causing errors